### PR TITLE
Move effect of climate control to the end of bodytemp calc

### PR DESCRIPTION
**Pull request by illi-kun**
First of all, function `void player::update_bodytemp()` is terribly huge and should be split to smaller functions. Its current state makes debugging operations harder than it should be.

This PR solves #14213. Previously, effect of climate control was applied before effect of blood loss, and it cause weird behavior described in the mentioned issue.

---

Simplified calcs steps:
- Initial wearing:

```
<...>
temp_conv = 6070;
blood_loss = 85; // in %
if (temp_conv > 6500) { // BODYTEMP_HOT = 6500
  temp_conv = 5000;     // BODYTEMP_NORM = 5000
}
temp_conv -= blood_loss * temp_conv / 200;
```

result is: 85 \* 6070 / 200 = 2580
- initial wearing + leg warmers:

```
<...>
temp_conv = 6070 + 500; // +500 due to warmers
blood_loss = 85; // in %
if (temp_conv > 6500) { // BODYTEMP_HOT = 6500
  temp_conv = 5000;     // BODYTEMP_NORM = 5000
}
temp_conv -= blood_loss * temp_conv / 200;
```

result is: 85 \* 5000 / 200 = 2125

---

I've moved the effect of climate control to the end of temperature calculations for avoiding such weird behavior in future.

Sorry for lack of explanations, it's 4am and I'm going to sleep right now ;-)

### CODE REVIEW

**Comment by kevingranade** [comment]
I totally agree about the size, in particular what I'd like to see is something that isolates the body temperature calculation function from the code to *find* all the pieces of data it needs, the "sensor" concept I added in #14240 might be suitable to do that.
